pkg/pillar/vault/handler_zfs.go

```diff
@@ -83,6 +90,20 @@ func (h *ZFSHandler) RemoveDefaultVault() error {
 // SetupDefaultVault setups vaults on zfs, using zfs native encryption support
 func (h *ZFSHandler) SetupDefaultVault() error {
 	if !etpm.IsTpmEnabled() {
+		if base.IsHVTypeKube() {
+			if zfs.DatasetExist(h.log, types.SealedDataset) {
+				return MountVaultZvol(h.log, types.SealedDataset)
+			}
+			if err := CreateZvolVault(h.log, types.SealedDataset, "", false); err != nil {
```

Comment:
Ditto. This function creates an ext4 filesystem in the zvol in the vault dataset.